src/Symfony/Component/HttpFoundation/Tests/IpUtilsTest.php

```diff
@@ -101,4 +101,25 @@ public function invalidIpAddressData()
             'invalid request IP with invalid proxy wildcard' => ['0.0.0.0', '*'],
         ];
     }
+
+    /**
+     * @dataProvider anonymizedIpData
+     */
+    public function testAnonymize($ip, $expected)
+    {
+        $this->assertSame($expected, IpUtils::anonymize($ip));
+    }
+
+    public function anonymizedIpData()
+    {
+        return [
+            ['192.168.1.1', '192.168.1.0'],
+            ['1.2.3.4', '1.2.3.0'],
+            ['2a01:198:603:0:396e:4789:8e99:890f', '2a01:198:603:0:396e:4789:8e99:0'],
+            ['0:0:0:0:0:0:0:1', '0:0:0:0:0:0:0:0'],
+            ['0:0:603:0:396e:4789:8e99:0001', '0:0:603:0:396e:4789:8e99:0'],
```

Comment:
For IPv6, maybe we should remove more bits, I am not sure what would be considered anonymous enough. If someone feels like researching what would be appropriate be my guest :)